packit_service/worker/helpers/sync_release/sync_release.py

```diff
@@ -57,7 +57,7 @@ def branches(self) -> Set[str]:
         Return all valid branches from config.
         """
         branches = get_branches(
-            *self.job.dist_git_branches, default=self.default_dg_branch
```

Comment:
@lbarcziova to make things work in my field test I had to change this initialization parameter, but I am not totally sure this is right. Just asking. Thanks!

Reply:
the `default` here should be correct, what was the issue when testing with this?

Reply:
This is my [`.packit.yaml`](https://src.fedoraproject.org/rpms/python-teamcity-messages/blob/rawhide/f/.packit.yaml) file in the `rawhide` branch in dist-git.
The default branch is `rawhide`.
If I do not change the above call, I can not work on dist-git. The code will look for a dist-git `default` like `main` or `master` (I don't remember) instead of `rawhide` and fails.

Reply:
ah, interesting. This code was not introduced by the `pull-from-upstream` changes; I think it was like that all the time. 

[Here](
https://github.com/packit/packit/blob/281a5acbdbc8358405bc75e72f568cb7fd7a8a84/packit/config/aliases.py#L152-L154) is the related code, but I am not that familiar with it. @lachmanfrantisek @jpopelka , from the git blame, do you think we should pass here also the `default_dg_branch` from API and not rely on the default `main` as e.g. repo linked by Maja doesn't include that branch?

Reply:
I saw that a lot of tests are failing now with my change, so I reverted it.
I am wondering if this [code](https://github.com/packit/packit/blob/281a5acbdbc8358405bc75e72f568cb7fd7a8a84/packit/config/aliases.py#L155-L156) could be related with my problem.
If I [use here](https://github.com/packit/packit-service/blob/3bea9d46865314c6e6a62a4cadcc4e0e808575d3/packit_service/worker/helpers/sync_release/sync_release.py#L59) the `with_aliases=True` flag then branch `rawhide` is used and everything goes smoothly.
I can also see that with the `with_aliases` flag less unit tests fail...

Reply:
from looking into the code, it would make sense to me to pass the `default_dg_branch` and not enable `with_aliases`, since `with_aliases` will result in getting both `main` and `rawhide`, but getting `main` doesn't make sense if it doesn't exist (your case)...but I would like someone to confirm this will be correct

Reply:
I would try to use the default branch of the Pagure repository so we are safe to use repository of any structure (even if the `main` branch misses because of an issue or for a good reason).

Reply:
> I would try to use the default branch of the Pagure repository so we are safe to use repository of any structure (even if the `main` branch misses because of an issue or for a good reason).

I tested my last commit on my weird repo config and it is working and no tests are failing. So probably it is safe to add it.